sdk/lib/collection/linked_hash_map.dart

```diff
@@ -14,11 +16,78 @@ part of dart.collection;
 /// but removing the key and adding it again
 /// will make it be last in the iteration order.
 ///
+/// **Notice:**
+/// It is generally not allowed to modify the map (add or remove keys) while
+/// an operation is being performed on the map, for example in functions called
+/// during a [forEach] or [putIfAbsent] call.
+/// Modifying the map while iterating the keys or values
+/// may also break the iteration.
+///
 /// The keys of a `LinkedHashMap` must have consistent [Object.==]
 /// and [Object.hashCode] implementations. This means that the `==` operator
 /// must define a stable equivalence relation on the keys (reflexive,
 /// symmetric, transitive, and consistent over time), and that `hashCode`
 /// must be the same for objects that are considered equal by `==`.
+///
+/// Example:
+///
+/// ```dart
+/// final equatorialDiameters = {};
+///
+/// // To add data to map, call addAll or addEntries
+/// equatorialDiameters
```

Comment:
For me these 0.949, 0.532, 11.209 keys are confusing. Equatorial Diameters is a fun idea, but I would rather use 1, 2, 3 for the key with the planet names. What rest of you think about this?

Reply:
I asked Iiro and he prefers those equatorial diameters numbers, so I guess it is a preference :D

Reply:
keeping, those for now..